include/range/v3/range/concepts.hpp

```diff
@@ -188,88 +178,28 @@ namespace ranges
     /// \cond
     namespace detail
     {
-        struct enable_view_helper_
-        {
-            bool result_;
-
-            template<typename T>
-            static constexpr auto test(T const *) -> CPP_ret(bool)( //
-                requires range<T> && range<T const>)
-            {
-                return RANGES_IS_SAME(iter_reference_t<iterator_t<T>>,
-                                      iter_reference_t<iterator_t<T const>>);
-            }
-            static constexpr auto test(void const *) -> bool
-            {
-                return true;
-            }
-            template<typename T>
-            constexpr enable_view_helper_(T const * p)
-              : result_(enable_view_helper_::test(p))
-            {}
-        };
-        constexpr bool enable_view_impl_(...)
-        {
-            return false;
-        }
-        constexpr bool enable_view_impl_(view_base const *)
-        {
-            return true;
-        }
-        constexpr bool enable_view_impl_(enable_view_helper_ ev)
-        {
-            return ev.result_;
-        }
-        template<typename T>
-        constexpr bool enable_view_impl_(std::initializer_list<T> const *)
-        {
-            return false;
-        }
-        template<typename Key, typename Compare, typename Alloc>
-        constexpr bool enable_view_impl_(std::set<Key, Compare, Alloc> const *)
-        {
-            return false;
-        }
-        template<typename Key, typename Compare, typename Alloc>
-        constexpr bool enable_view_impl_(std::multiset<Key, Compare, Alloc> const *)
-        {
-            return false;
-        }
-        template<typename Key, typename Hash, typename Pred, typename Alloc>
-        constexpr bool enable_view_impl_(
-            std::unordered_set<Key, Hash, Pred, Alloc> const *)
-        {
-            return false;
-        }
-        template<typename Key, typename Hash, typename Pred, typename Alloc>
-        constexpr bool enable_view_impl_(
-            std::unordered_multiset<Key, Hash, Pred, Alloc> const *)
-        {
-            return false;
-        }
-        // BUGBUG TODO
-        // template<typename BidiIter, typename Alloc>
-        // constexpr bool enable_view_impl_(std::match_results<BidiIter, Alloc> const *)
-        // {
-        //     return false;
-        // }
         template<typename T>
-        constexpr T const * nullptr_(int)
-        {
-            return nullptr;
-        }
-        template<typename T>
-        constexpr int nullptr_(long)
-        {
-            return 0;
-        }
+        struct enable_view_
+          : std::is_base_of<view_base, T>
```

Comment:
Why `is_base_of` instead of `derived_from`?

Reply:
I am not opposed to folks privately inheriting from `view_base`. That's good enough for me.